composeApp/src/commonMain/kotlin/com/daedan/festabook/data/repository/FestivalNotificationRepositoryImpl.kt

```
@@ -3,38 +3,45 @@ package com.daedan.festabook.data.repository
 ) : FestivalNotificationRepository {
     override suspend fun saveFestivalNotification(): Result<Unit> {
-        val deviceId = deviceLocalDataSource.getDeviceId()
-        if (deviceId == null) {
-//            Timber.e("${::FestivalNotificationRepositoryImpl.name}: DeviceId가 없습니다.")
-            return Result.failure(IllegalStateException())
-        }
-        val festivalId = festivalLocalDataSource.getFestivalId()
+        val deviceId =
+            deviceLocalDataSource.getDeviceId().first() ?: run {
+//            Timber.e("${this::class.simpleName}: DeviceId가 null 입니다.")
+                return Result.failure(IllegalStateException())
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)